internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -189,14 +210,15 @@ func (a *runTaskAction) Do(o *Orchestrator) error {
 	}()
 
 	if taskID == 1 {
+		o.curTask = a.task
```

Comment:
Because we want to ensure that there are no `pause` container changes incase if the `taskId` is other than 1.
and also at the same time `o.currTask` is passed as parameter to function `o.stopTask`

Reply:
makes sense! I wonder if something like ⬇️ would help clarify the logic:
```go
func (a *runTaskAction) Do(o *Orchestrator) error {
   prevTask := o.currTask
   o.currTask = a.Task
   if taskID == 1 {
   } else {
       prevOpts := o.pauseRunOptions(prevTask)
       // ...
       if err := o.stopTask(ctx, prevTask); err != nil {
			return fmt.Errorf("stop existing task: %w", err)
		}
   }
}
```